### PR TITLE
ColorModal UI improvements + fix reset

### PR DESCRIPTION
**Pull request by lanzhenw**
- [x] If dataset has default setting, reset session color should reset to dataset default. 
- [x] Make color modal resizable. 
- [x] Menu list selected item highlighted
- [x] Update texts
- [x] Improve UI in field settings
- [x] Fix a bug with rendering color by index
- [x] Update on overlay on looker3d

https://github.com/voxel51/fiftyone/assets/17770824/ab659f53-a6e5-4535-81e0-70846af851c0

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/state/src/hooks/useClearSessionColorScheme.ts [hunk 1]
```diff
@@ -19,17 +20,31 @@ const useClearSessionColorScheme = () => {
   const setSessionColorSchemeState = useSetRecoilState(sessionColorScheme);
   const dataset = useRecoilValue(datasetName);
   const stages = useRecoilValue(view);
+  const defaultSetting = useRecoilValue(datasetAppConfig).colorScheme;
 
   function onClear(saveToApp: boolean) {
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useClearSessionColorScheme.ts:25]: `useMemo` could be good to use here

File: app/packages/state/src/hooks/useClearSessionColorScheme.ts [hunk 2]
```diff
@@ -19,17 +20,31 @@ const useClearSessionColorScheme = () => {
...
+        defaultSetting && !saveToApp
+          ? defaultSetting.colorPool
+          : DEFAULT_APP_COLOR_SCHEME.colorPool,
+      customizedColorSettings:
+        defaultSetting?.customizedColorSettings && !saveToApp
+          ? defaultSetting.customizedColorSettings
+          : null,
     };
-
     setSessionColorSchemeState(combined);
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useClearSessionColorScheme.ts:48]: Small detail, but usually I put state updates like this in the `onComplete` callback of the mutation to ensure the API call was successful and things are synced

**Review by benjaminpkane — APPROVED** [review]
LGTM 👍

File: app/packages/core/src/components/Sidebar/Sidebar.module.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .resizeHandle:hover,
 .resizeHandle:active,
 .resizeHandle:focus {
-  background: #007fd4;
+  background: #ff6d05;
```
[the remark this answers is not in the document]
Comment (lanzhenw) [reply app/packages/core/src/components/Sidebar/Sidebar.module.css:4]: Added our primary theme color as `--fo-palette-voxel-500` in the App.css. Does this name make sense?

[the remark this answers is not in the document]
Comment (benjaminpkane) [reply app/packages/core/src/components/Sidebar/Sidebar.module.css:4]: Ok, if it is just the brand color, `var(--joy-palette-primary-main)` should probably be used

File: app/packages/state/src/hooks/useClearSessionColorScheme.ts (hunk 1 above)
Comment (lanzhenw) [reply app/packages/state/src/hooks/useClearSessionColorScheme.ts:25]: "useRecoilValue" cannot be called inside the useMemo callback.

File: app/packages/core/src/components/ColorModal/ColorModal.tsx [hunk 4]
```diff
@@ -41,27 +68,75 @@ const ColorModal = () => {
...
+                ["top"]: { top: 0, height: 4 },
+                ["bottom"]: { bottom: 0, height: 4 },
+              }}
+              handleClasses={{
+                ["right"]: resizeHandle,
+                ["left"]: resizeHandle,
+                ["top"]: resizeHandle,
+                ["bottom"]: resizeHandle,
+              }}
+              // style={{ overflow: "auto" }}
```
Comment (imanjra) [inline app/packages/core/src/components/ColorModal/ColorModal.tsx:106]: can comment be removed?

File: app/packages/core/src/components/ColorModal/ColorModal.tsx [hunk 5]
```diff
@@ -41,27 +68,75 @@ const ColorModal = () => {
...
+                setWidth(width + dw);
+                setHeight(height + dh);
+                // reset sidebar width on double click
+                if (e.detail === 2) {
+                  setWidth(860);
+                  setHeight(680);
+                }
+              }}
+              handleStyles={{
+                ["right"]: { right: 0, width: 4 },
```
Comment (imanjra) [inline app/packages/core/src/components/ColorModal/ColorModal.tsx:95]: suggestion for future (may ignore): we should move these to a common util. I think it's duplicated in couple placed

File: app/packages/core/src/components/ColorModal/ColorModal.tsx [hunk 6]
```diff
@@ -41,27 +68,75 @@ const ColorModal = () => {
...
+                ["left"]: { left: 0, width: 4 },
+                ["top"]: { top: 0, height: 4 },
+                ["bottom"]: { bottom: 0, height: 4 },
+              }}
+              handleClasses={{
+                ["right"]: resizeHandle,
```
Comment (imanjra) [inline app/packages/core/src/components/ColorModal/ColorModal.tsx:101]: nice

File: app/packages/state/src/hooks/useClearSessionColorScheme.ts (hunk 2 above)
Comment (lanzhenw) [reply app/packages/state/src/hooks/useClearSessionColorScheme.ts]: Yes, just added it. Thanks for catching that!

File: app/packages/core/src/components/ColorModal/ColorModal.tsx [hunk 7]
```diff
@@ -41,27 +68,75 @@ const ColorModal = () => {
...
-                  {field === ACTIVE_FIELD.json && <JSONViewer />}
-                  {typeof field !== "string" && field && (
-                    <FieldSetting field={activeColorModalField as Field} />
-                  )}
-                </Display>
-              </DraggableContent>
-              <ColorFooter />
-            </Container>
+            <Resizable
+              size={{ height, width }}
```
Comment (imanjra) [inline app/packages/core/src/components/ColorModal/ColorModal.tsx:72]: suggestion for future (may ignore):  Looks like we're duplicating most of these props. We should probably create re-usable wrapper

**Pull request merged**